pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1561,6 +1562,7 @@ func (c *Cloud) DetachDisk(diskName KubernetesVolumeID, nodeName types.NodeName)
 		return "", errors.New("no response from DetachVolume")
 	}
 
+	// TODO-P1: If we are doing concurrent detach operations, this is very expensive... bring in the cache layer.
 	attachment, err := disk.waitForAttachmentStatus("detached")
```

Comment:
We are only allowing concurrent detach operations for different volumes, each doing its own polling with volume-name as query.

agreed though - having a coherent cache with all "known" volumes will eliminate the need for polling individual volumes.